openproblems/tasks/_batch_integration/batch_integration_graph/methods/saucie_embed.py

```diff
@@ -0,0 +1,92 @@
+# from ....tools.normalize import log_cpm
+from .....tools.decorators import method
+from .....tools.utils import check_version
+
+
+@method(
+    method_name="Saucie embedding output",
+    paper_name="Exploring single-cell data with deep multitasking neural networks",
+    paper_url="https://www.nature.com/articles/s41592-019-0576-7",
+    paper_year=2019,
+    code_url="https://github.com/KrishnaswamyLab/SAUCIE",
+    code_version=check_version("SAUCIE"),
+    image="openproblems-python37-saucie",  # only if required
+)
+def saucie_embed_full_unscaled(adata):
+    from scib.integration import runSaucie
+    from scib.preprocessing import reduce_data
+
+    adata.obs["batch"] = adata.obs["batch"].astype("category")
+    adata = runSaucie(adata, "batch")
+    reduce_data(adata, umap=False, use_rep="X_emb")
+    # Complete the result in-place
+    adata.var = adata.var.iloc[:, :2]
```

Comment:
what is this line?

Reply:
saucie was adding some duplicate columns to var which was breaking adata.write()